### ISSUE TITLE
Replace flatpickr

### ISSUE THREAD

**Issue by formfcw** [directus/directus#25341]
### Describe the Bug

The flatpickr library used in the `<v-date-picker>` component is outdated, and should be replaced with a library that offers the same or better features. It should support a11y.

### To Reproduce

https://github.com/flatpickr/flatpickr/discussions/2479

### Directus Version

v11.8.0

### Hosting Strategy

Self-Hosted (Docker Image)
